src/app/games/trexo/tests/trexo.component.spec.ts

```diff
@@ -0,0 +1,186 @@
+            // Given any board on which a first click has been made
+            await testUtils.expectClickSuccess('#space_5_5');
+
+            // When clicking on a far away coord (that is still valid)
+            // Then the click should be a success
+            await testUtils.expectClickSuccess('#space_7_7');
+            // And the dropped piece changed
+            testUtils.expectElementNotToExist('#dropped_piece_5_5_0');
+            testUtils.expectElementToExist('#dropped_piece_7_7_0');
+        }));
+        it(`should show last move`, fakeAsync(async() => {
+            // Given any board on which a first click has been made
+            await testUtils.expectClickSuccess('#space_5_5');
+
+            // When finalizing the move
+            const move: TrexoMove = TrexoMove.from(new Coord(4, 5), new Coord(5, 5)).get();
+            await testUtils.expectMoveSuccess('#space_4_5', move);
+
+            // Then the dropped coords should be highlighted
+            testUtils.expectElementToHaveClass('#tile_4_5_0', 'last-move-stroke');
+            testUtils.expectElementToHaveClass('#tile_5_5_0', 'last-move-stroke');
+        }));
+        it(`should cancel move when clicking again on the same coord`, fakeAsync(async() => {
+            // Given any board on which a first click has been made
+            await testUtils.expectClickSuccess('#space_5_5');
+
+            // When clicking on the same coord again
+            // Then the move should be cancelled without toast
+            await testUtils.expectClickSuccess('#space_5_5');
+            // And the piece deselected
+            testUtils.expectElementNotToExist('#dropped_piece_5_5');
+        }));
+        it('should highlight victory', fakeAsync(async() => {
+            // Given any board on which a 4 moves have already been done, aligning piece of Player.ZERO
+            await doMove(new Coord(3, 3), new Coord(3, 2));
+            await doMove(new Coord(4, 3), new Coord(4, 4));
+            await doMove(new Coord(5, 3), new Coord(5, 2));
+            await doMove(new Coord(6, 3), new Coord(6, 4));
+            await testUtils.expectClickSuccess('#space_7_2');
+
+            // When doing the victorious move
+            const move: TrexoMove = TrexoMove.from(new Coord(7, 3), new Coord(7, 2)).get();
+
+            // Then the 5 victory coords should be highlighted
+            await testUtils.expectMoveSuccess('#space_7_3', move);
+            testUtils.expectElementToHaveClass('#tile_3_3_0', 'victory-stroke');
+            testUtils.expectElementToHaveClass('#tile_4_3_0', 'victory-stroke');
+            testUtils.expectElementToHaveClass('#tile_5_3_0', 'victory-stroke');
+            testUtils.expectElementToHaveClass('#tile_6_3_0', 'victory-stroke');
+            testUtils.expectElementToHaveClass('#tile_7_3_0', 'victory-stroke');
+        }));
+    });
+    describe(`view`, () => {
```

Comment:
techniquement faudrait répéter tous les tests précédents dans l'autre vue également ! (peut-être simplement faire une boucle qui clique un coup sur le bouton, un coup pas, et qui fait tous les describe/it précédents)

Reply:
Voilà, j'ai techniquement loopé !
Tiens, j'ai voulu écrire "louper" c'qui aurais été un faux-amis frenglisant.
Et ceci m'as fait penser à un collègue qui utilisé comme si c'était un verbe francophone le vers "to ship" donc "il a chippé les commandes!" :D